server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java

```diff
@@ -102,34 +103,46 @@ public void setMappingLimit(int mappingLimit) throws IllegalArgumentException {
 
     @Override
     public Mappings getResolvedMappings(String user, Domain domain) throws ErrorMappingException, RecipientRewriteTableException {
-        return getMappings(Username.fromLocalPartWithDomain(user, domain), mappingLimit);
+        return getMappings(Username.fromLocalPartWithDomain(user, domain), mappingLimit, Optional.empty());
     }
 
-    private Mappings getMappings(Username username, int mappingLimit) throws ErrorMappingException, RecipientRewriteTableException {
+
+    @Override
+    public Mappings getResolvedMappings(String user, Domain domain, EnumSet<Type> mappingTypes) throws ErrorMappingException, RecipientRewriteTableException {
+
+        return getMappings(Username.fromLocalPartWithDomain(user, domain), mappingLimit, Optional.of(mappingTypes));
+    }
+
+    private Mappings getMappings(Username username, int mappingLimit, Optional<EnumSet<Type>> mappingTypes) throws ErrorMappingException, RecipientRewriteTableException {
```

Comment:
the use of optional here is too complex: with EnumSet, you can say `allOf` which is what you want.
There's no need for an optional list of types.